app/src/main/java/org/greenstand/android/TreeTracker/viewmodels/DataViewModel.kt

```diff
@@ -84,45 +86,53 @@ class DataViewModel(private val dao: TreeTrackerDAO,
 
     fun sync() {
         viewModelScope.launch {
+            val treeData = loadPlanterAccountData().treeData
+            val totalCount = treeData.uploadedCount + treeData.waitingToUploadCount
             if (_isSyncing == null || _isSyncing == false) {
                 val treesToSync =
                     withContext(Dispatchers.IO) { dao.getNonUploadedTreeCaptureCount() }
-                when (treesToSync) {
-                    0 -> toastLiveData.value = R.string.nothing_to_sync
-                    else -> startDataSynchronization()
+                if (treesToSync == 0) {
+                    toastLiveData.value = R.string.nothing_to_sync
+                    isSyncingLiveData.value = false
+                } else {
+                    isSyncingLiveData.value = true
+                    startDataSynchronization()
                 }
-                val (total, synced, waiting) = loadTreeInfo()
-                analytics.syncButtonTapped(total, synced, waiting)
+                analytics.syncButtonTapped(totalCount, treeData.uploadedCount, treeData.waitingToUploadCount)
             } else {
                 updateTimerJob?.cancel()
                 updateTimerJob = null
                 workManager.cancelUniqueWork(TreeSyncWorker.UNIQUE_WORK_ID)
                 syncNotification.removeNotification()
-
-                val (total, synced, waiting) = loadTreeInfo()
-                analytics.stopButtonTapped(total, synced, waiting)
+                isSyncingLiveData.value = false
+                analytics.stopButtonTapped(totalCount, treeData.uploadedCount, treeData.waitingToUploadCount)
             }
         }
     }
 
-
-    private suspend fun loadTreeInfo(): TreeData {
+    private suspend fun loadPlanterAccountData(): PlanterAccountData {
```

Comment:
This function has gotten very large, can we please move this logic into a "usecase" class that returns the PlanterAccountData?

Reply:
The function isn't as big compared to sync function in this class. Since this function is simply accessing data and instantiating a PlanterAccountData it could be part of a Repository so that it could be used by multiple view models if there is a need for it. Right now there is not any usage of that sort I believe.

It feels to me that this method has no sufficient business logic/use case to deserve a use case.  For my clarification, Use case classes I assume are orchestraters of other domain classes to effect a use case, correct?